zebra-rpc/src/errors.rs

```diff
@@ -0,0 +1,9 @@
+use jsonrpc_core::types::error::{Error, ErrorCode};
+
+pub(crate) fn make_server_error(message: impl std::fmt::Display) -> Error {
```

Comment:
I think we should put it behind a feature flag for now and do the replacements after the audit.

Updated in [`dd181c1`](https://github.com/ZcashFoundation/zebra/pull/5526/commits/dd181c1c33b6d3d0842d6c6452bc20cc184a1d15) along with a `TODO` and doc comment.

Reply:
I think this might be easier to use as a trait, so we can write code like:
```rust
ready().await.map_server_error()?;

get_height().ok_or_server_error("No blocks in state")?;
```

Something like this, but maybe with an `error_code: Into<Option<i64>>` argument.

```rust
impl<T, E> MapServerError for Result<T, E> where E: Debug {
    fn map_server_error(result: Self) -> Result<T, jsonrpc_core::Error> {
        result.map_err(|error| jsonrpc_core::Error {
            code: ErrorCode::ServerError(0),
            message: format!("{error:?}"),
            data: None,
        })
    }
}

impl<T> OkOrServerError for Option<T> {
    fn ok_or_server_error<S: ToString>(option: Self, message: S) -> Result<T, jsonrpc_core::Error> {
        option.ok_or(jsonrpc_core::Error {
            code: ErrorCode::ServerError(0),
            message: message.to_string(),
            data: None,
        })
    }
}
```

Can we do this refactor in a separate PR?
It seems unrelated to the topic of this PR.

Reply:
I removed `make_server_error` for now because on second thought it's inconsistent and that's much better.